src/impl/tokenParser.js

```diff
@@ -64,32 +69,32 @@ function unitForToken(token, loc) {
           return oneOf(loc.eras("long", false), 0);
         // years
         case "y":
-          return intUnit(/\d{1,6}/);
+          return intUnit(oneToSix);
         case "yy":
           return intUnit(twoToFour, untruncateYear);
         case "yyyy":
           return intUnit(four);
         case "yyyyy":
-          return intUnit(/\d{4,6}/);
+          return intUnit(fourToSix);
         case "yyyyyy":
-          return intUnit(/\d{6}/);
+          return intUnit(six);
         // months
         case "M":
           return intUnit(oneOrTwo);
         case "MM":
           return intUnit(two);
         case "MMM":
-          return oneOf(loc.months("short", false, false), 1);
```

Comment:
I don't see why these should change from false to true?

Reply:
It is to bring parity with format string in `toFormat`. For instance, let us consider the locale to be `zh` and numberingSystem to be `hanidec`. It would be something like this with the above fix:

<img width="1304" alt="Screenshot 2019-04-06 at 10 31 23 AM" src="https://user-images.githubusercontent.com/189344/55665114-408ea500-5857-11e9-8bc1-3fb8cdd1f01c.png">

Earlier, `loc.months` returned an array of months in a different format which made it difficult to easily transpose the output of `toFormat` with `fromFormat`. The ideal situation would be that the user can feed the output of `toFormat` to `fromFormat` and get back the original value. I know that it doesn't work in some cases (like if the format has `EEEEE` making it ambiguous. does "T" stand for "Tuesday" or "Thursday"?).

From my tests, the output of:

`loc.months("long", false, false)` for `hanidec` numberingSystem was:

`[ '一月', '二月', '三月', '四月', '五月', '六月', '七月', '八月', '九月', '十月', '十一月', '十二月' ]`

As you can clearly see the `toFormat` just output the numerical equivalent of "April" as `四` but `loc.months` output the string equivalent of "April" as `四月`. The parser was unable to parse the string (because there is no `四月` in the localized string) causing it to output `null` for rawMatches. The correct output for `loc.months` should be:

`[ '一', '二', '三', '四', '五', '六', '七', '八', '九', '一〇', '一一', '一二' ]`

I think your intent was right but the flags were probably flipped in the implementation of `Locale.months` in locale.js. I was not sure about it so I did not touch the locale.js code. I instead flipped the boolean in the parser logic itself. It did not cause any other tests to fail so I am assuming it wasn't tested earlier and my tests were the first to check that line of code.

Nevertheless please do let me know if my understanding was incorrect. I am new to your library (which is absolutely amazing by the way) and am learning the internals on the go. 😃